### PR TITLE
abinit: add v9.10.3

### PR DESCRIPTION
**Pull request by downloadico**
changed WANNIER90_CPPFLAGS for 9.10.3
changed variable name for fortran compiler from FC to F90 for 9.10.3

### CODE REVIEW

File: var/spack/repos/builtin/packages/abinit/package.py [hunk 1]
```diff
@@ -152,7 +154,7 @@ def configure_args(self):
                 oapp("--with-wannier90-bins={0}".format(spec["wannier90"].prefix.bin))
                 oapp("--enable-connectors")
                 oapp("--with-dft-flavor=atompaw+libxc+wannier90")
-            else:
+            elif "@:9.8" in spec:
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/abinit/package.py]: Version range checks should be checked with `spec.satisfies()`.  See docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#testing-spec-constraints.

Please correct here and at least line 187.

File: var/spack/repos/builtin/packages/abinit/package.py [hunk 2]
```diff
@@ -161,6 +163,14 @@ def configure_args(self):
                         ),
                     ]
                 )
+            else:
+                options.extend(
+                    [
+                        "WANNIER90_CPPFLAGS=-I{0}".format(spec["wannier90"].prefix.modules),
+                        "WANNIER90_LIBS=-L{0}".format(spec["wannier90"].prefix.lib),
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/abinit/package.py]: We are slowly replacing `format()` with f-strings (https://docs.python.org/3/tutorial/inputoutput.html) so suggested using that approach here and lines 188 and 190.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the version sha256 but have some suggestions for version range checks and use of f-strings.

File: var/spack/repos/builtin/packages/abinit/package.py (hunk 1 above)
Comment (downloadico) [reply var/spack/repos/builtin/packages/abinit/package.py]: Should be corrected now

**Review by tldahlgren — DISMISSED** [review]
LGTM

**Comment by tldahlgren** [comment]
Just noticed there are conflicts that need to be resolved.

**Comment by downloadico** [comment]
I'll submit a new PR for the changes I'd like

**Pull request closed without merging**